src/_gettsim/transfers/arbeitsl_geld_2/sgb_ii_bedarf.py

```diff
@@ -0,0 +1,275 @@
+    elif alter <= anteile[5]["max_alter"] and gleiche_fg_kindergeldempfänger_kind:
+        out = regelsatz * anteile[5]["anteil"]
+    elif alter <= anteile[4]["max_alter"] and gleiche_fg_kindergeldempfänger_kind:
+        out = regelsatz * anteile[4]["anteil"]
+    else:
+        out = 0.0
+
+    return float(out)
+
+
+@policy_info(start_date="2011-01-01", name_in_dag="arbeitsl_geld_2_kindersatz_m")
+def arbeitsl_geld_2_kindersatz_m_ab_2011(
+    alter: int,
+    gleiche_fg_kindergeldempfänger_kind: bool,
+    arbeitsl_geld_2_params: dict,
+) -> float:
+    """Basic monthly subsistence / SGB II needs of children since 2011.
+
+    Note: Since 2023, Arbeitslosengeld 2 is referred to as Bürgergeld.
+
+    Parameters
+    ----------
+    alter
+        See basic input variable :ref:`alter`.
+    gleiche_fg_kindergeldempfänger_kind
+        See :func:`gleiche_fg_kindergeldempfänger_kind`.
+    arbeitsl_geld_2_params
+        See params documentation :ref:`arbeitsl_geld_2_params <arbeitsl_geld_2_params>`.
+
+    Returns
+    -------
+    SGB II needs of child
+
+    """
+
+    kindersofortzuschl = arbeitsl_geld_2_params.get("kindersofortzuschl", 0.0)
+
+    if (
+        alter <= arbeitsl_geld_2_params["regelsatz"][6]["max_alter"]
+        and gleiche_fg_kindergeldempfänger_kind
+    ):
+        out = arbeitsl_geld_2_params["regelsatz"][6]["bedarf"] + kindersofortzuschl
+    elif (
+        alter <= arbeitsl_geld_2_params["regelsatz"][5]["max_alter"]
+        and gleiche_fg_kindergeldempfänger_kind
+    ):
+        out = arbeitsl_geld_2_params["regelsatz"][5]["bedarf"] + kindersofortzuschl
+    elif (
+        alter <= arbeitsl_geld_2_params["regelsatz"][4]["max_alter"]
+        and gleiche_fg_kindergeldempfänger_kind
+    ):
+        out = arbeitsl_geld_2_params["regelsatz"][4]["bedarf"] + kindersofortzuschl
+    elif gleiche_fg_kindergeldempfänger_kind:  # adult children with parents in FG
+        out = arbeitsl_geld_2_params["regelsatz"][3] + kindersofortzuschl
+    else:
+        out = 0.0
+
+    return float(out)
+
+
+@policy_info(end_date="2010-12-31", name_in_dag="arbeitsl_geld_2_erwachsenensatz_m")
+def arbeitsl_geld_2_erwachsenensatz_bis_2010_m(
+    _arbeitsl_geld_2_alleinerz_mehrbedarf_m: float,
+    arbeitsl_geld_2_kindersatz_m: float,
+    p_id_einstandspartner: int,
+    arbeitsl_geld_2_params: dict,
+) -> float:
+    """Basic monthly subsistence / SGB II needs for adults without dwelling.
+
+    Parameters
+    ----------
+    _arbeitsl_geld_2_alleinerz_mehrbedarf_m
+        See :func:`_arbeitsl_geld_2_alleinerz_mehrbedarf_m`.
+    arbeitsl_geld_2_kindersatz_m
+        See :func:`arbeitsl_geld_2_kindersatz_m`.
+    p_id_einstandspartner
+        See basic input variable :ref:`p_id_einstandspartner`.
+    arbeitsl_geld_2_params
+        See params documentation :ref:`arbeitsl_geld_2_params <arbeitsl_geld_2_params>`.
+
+    Returns
+    -------
+
+    """
+    if p_id_einstandspartner > 0:  # BG with 2 adults
```

Comment:
I'm afraid I can't quite follow here. It's true that the comments are confusing (because they refer to the BG composition while the function is on individual level.)

The first if condition checks whether the obs has an Einstandspartner. Given the correct `bg_id` this implies that both of them are in the same BG and, hence, each of them should get half of the couple Regelbedarf.

The second if condition checks whether the obs has a claim on the Kinderregelsatz. If not, the obs must be the only adult in the BG (and potentially a single parent).

I don't see how this function may create wrong output, but the object in your sentence "the regelsatz of the is returned" is missing, so maybe I didn't get your point. 😄 

The output of `regelsatz_fg_with_2_adults_1_kid_in_1_bg` looks fine to me (note that the test's targets are on BG level, while the function is on individual level)

Reply:
Great, thanks. Indeed I interpreted the comment ("BG with two adults") more than the code, I think. That mutilated sentence should have included a "adults".

Do we have a check for the `einstandspartner` / `bg_id` - relationship in that data check list? Just in case a user seems to ignore that.

Thanks for pointing out that the tests are at the `bg` level. I did not realise that. They should move to the individual level then!